File: backend/src/controllers/auth.controller.ts
import { authenticate } from '@loopback/authentication';
import { inject } from '@loopback/core';
import { get } from '@loopback/rest';
import { SecurityBindings, UserProfile } from '@loopback/security';

export class AuthController {
  constructor() {}

  @get('/auth/currentUser')
  @authenticate('firebase')
  async currentUser(@inject(SecurityBindings.USER, { optional: true }) user: UserProfile): Promise<UserProfile> {
    return user;
  }
}

File: backend/src/main/kotlin/com/ynixt/sharedfinances/controllers/rest/AuthController.kt
package com.ynixt.sharedfinances.controllers.rest

import com.ynixt.sharedfinances.mapper.UserMapper
import com.ynixt.sharedfinances.model.dto.user.CurrentUserDto
import com.ynixt.sharedfinances.service.SecurityService
import com.ynixt.sharedfinances.service.UserService
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class AuthController(
    private val userService: UserService,
    private val userMapper: UserMapper,
    private val securityService: SecurityService
) {
    @GetMapping("current-user")
    fun login(authentication: Authentication): CurrentUserDto {
        val user = securityService.authenticationToUser(authentication)!!
        return userMapper.toCurrentUserDto(userService.getForCurrentUser(user.id!!)!!)
    }
}
